client/src/main/resources/process-0.1.xsd

```diff
@@ -331,6 +338,40 @@
             <xs:enumeration value="oozie"/>
             <xs:enumeration value="pig"/>
             <xs:enumeration value="hive"/>
+            <xs:enumeration value="spark"/>
+        </xs:restriction>
+    </xs:simpleType>
+
+    <xs:complexType name="spark-attributes">
+        <xs:annotation>
+            <xs:documentation>
+                List of Spark attributes to run Spark application:
+                    master element indicates the url of the Spark Master. Ex: client, yarn-master, or local.
+                    mode element if present indicates the mode of spark, where to run spark driver program. Ex: client,cluster.
+                    name element indicates the name of the spark application.
+                    application-type indicates type of the spark application java, python.
+                    class element if present, indicates the spark's application main class.
+                    jar element indicates a comma separated list of jars or python files.
+                    spark-opts element if present, contains a list of spark options that can be passed to spark driver.
+                    arg element if present, contains arguments that can be passed to spark application.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:element name="master" type="xs:string" minOccurs="1" maxOccurs="1"/>
```

Comment:
Spark gives flexibility to the user to run application by specifying any value with master option like local, yarn-client, yarn-cluster or spark:// url . So if value to master specified through cluster xsd, then user will not be able to change the value of master option easily with different type of Spark application defined in process xsd.